runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/config/ConfigWriter.java

```diff
@@ -79,14 +108,31 @@ private void write0(
                 .withProvider(provider)
                 .withConfig(config)
                 .build();
-
             String jsonText = jsonb.toJson(namespace, NamespaceConfig.class);
-            JsonNode json = new ObjectMapper().readTree(jsonText);
+
+            JsonObject jsonObject = provider.createReader(new StringReader(jsonText)).readObject();
+            JsonObject patched = patch.apply(jsonObject);
+            StringWriter patchedText = new StringWriter();
+            JsonWriter jsonWriter = provider.createWriter(patchedText);
+            jsonWriter.write(patched);
+            String patchedJson = patchedText.toString();
+
+            JsonNode json = new ObjectMapper().readTree(patchedJson);
             YAMLMapper mapper = YAMLMapper.builder()
                 .disable(WRITE_DOC_START_MARKER)
                 .enable(MINIMIZE_QUOTES)
                 .build();
-            mapper.writeValue(writer, json);
+            String yaml = mapper.writeValueAsString(json);
+
+            for (String envVar : unquotedEnvVars)
+            {
+                yaml = yaml.replaceAll(
+                    Pattern.quote(String.format("\"${{env.%s}}\"", envVar)),
+                    String.format("\\${{env.%s}}", envVar)
```

Comment:
Btw, is this only needed because `MINIMIZE_QUOTES` is not removing the quotes generically?
Can we check why that's not already happening, since it is not typically necessary to have quotes on strings in yaml values.

Reply:
My observation is that it quotes the strings that contain curly braces.